src/libcollections/lib.rs

```diff
@@ -23,6 +23,8 @@
        html_root_url = "http://doc.rust-lang.org/nightly/",
        html_playground_url = "http://play.rust-lang.org/")]
 
+#![allow(trivial_cast)]
+#![allow(trivial_numeric_cast)]
```

Comment:
The lint names don't follow the conventions. `trivial_cast` -> `trivial_casts`, `trivial_numeric_cast` -> `trivial_numeric_casts`?